src/components/helpers/Audio.js

```diff
@@ -1,17 +1,30 @@
 import { Audio } from 'expo';
-import { forEach, map } from 'ramda';
+import { addIndex, all, forEach, filter, identity, map } from 'ramda';
 import { Component } from 'react';
 
+const mapIndexed = addIndex(map);
+
 export const createLoadSounds = Audio => {
   class LoadSounds extends Component {
     constructor(props) {
       super(props);
 
-      this.sounds = map(source => {
-        const sound = new Audio.Sound();
-        sound.loadAsync(source);
-        return sound;
-      }, this.props.sounds || []);
+      const sounds = filter(identity, this.props.sounds || []);
+
+      this.sounds = map(source => new Audio.Sound(), sounds);
+      this.state = { soundsLoaded: map(() => false, sounds) };
+    }
+
+    componentDidMount() {
+      mapIndexed((sound, i) => {
+        sound.loadAsync(this.props.sounds[i]).then(() => {
```

Comment:
Does this work? You filtered `this.props.sounds` to get the `this.sound` list, so I reckon the indices won't match up, or do they?

Reply:
Good catch, I'll change that